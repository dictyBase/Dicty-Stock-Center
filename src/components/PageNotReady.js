// @flow
import React from "react"
import { Link } from "react-router-dom"
import { withStyles } from "@material-ui/core/styles"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { Container, PrimaryButton } from "styles"

const styles = theme => ({
  jumbotron: {
    paddingTop: "30px",
    paddingBottom: "30px",
    marginBottom: "30px",
    color: "inherit",
    backgroundColor: "#eee",
    textAlign: "center",
  },
})

type Props = {
  classes: Object,
}

/**
 * General 404 error page, currently designated as "Page Not Ready"/"Under Construction"
 */

const PageNotReady = (props: Props) => {
  const { classes } = props

  return (
    <Container>
      <div className={classes.jumbotron}>
        <h1>
          <FontAwesomeIcon icon="wrench" /> Under Construction
        </h1>
        <br />
        <p>This page is not ready yet.</p>
        <br />
        <br />
        <Link to="/">
          <PrimaryButton>Stock Center Home</PrimaryButton>
        </Link>
      </div>
    </Container>
  )
}

export default withStyles(styles)(PageNotReady)
