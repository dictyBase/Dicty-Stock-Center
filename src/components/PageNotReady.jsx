// @flow
import React from "react"
import { Link } from "react-router-dom"
import FontAwesome from "react-fontawesome"
import { Container, Jumbotron, PrimaryButton } from "styles"

/**
 * General 404 error page, currently designated as "Page Not Ready"/"Under Construction"
 */

const PageNotReady = () => {
  return (
    <Container>
      <Jumbotron>
        <h1>
          <FontAwesome name="wrench" /> Under Construction
        </h1>
        <p>This page is not ready yet.</p>
        <PrimaryButton>
          <Link to="/">Stock Center Home</Link>
        </PrimaryButton>
      </Jumbotron>
    </Container>
  )
}

export default PageNotReady
