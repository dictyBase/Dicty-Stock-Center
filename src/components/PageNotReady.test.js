import React from 'react'
import renderer from 'react-test-renderer'
import PageNotReady from './PageNotReady'

test('matching a snapshot of PageNotReady', () => {
    const component = renderer.create(
    <PageNotReady />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
