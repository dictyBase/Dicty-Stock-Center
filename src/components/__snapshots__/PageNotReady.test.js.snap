// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of PageNotReady 1`] = `
<div
  className="sc-bdVaJa jqSxTV"
>
  <div
    className="sc-EHOje dVUPbG"
  >
    <h1>
      <span
        aria-hidden={true}
        className="fa fa-wrench"
      />
       Under Construction
    </h1>
    <p>
      This page is not ready yet.
    </p>
    <button
      className="sc-jAaTju ddurCY"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Stock Center Home
      </a>
    </button>
  </div>
</div>
`;
